hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -462,29 +465,47 @@ public List<ContainerInfo> listContainer(long startContainerID,
       final ContainerID containerId = ContainerID.valueOf(startContainerID);
       if (state != null) {
         if (factor != null) {
-          return scm.getContainerManager().getContainers(state).stream()
-              .filter(info -> info.containerID().getId() >= startContainerID)
-              //Filtering EC replication type as EC will not have factor.
-              .filter(info -> info
-                  .getReplicationType() != HddsProtos.ReplicationType.EC)
-              .filter(info -> (info.getReplicationFactor() == factor))
-              .sorted().limit(count).collect(Collectors.toList());
+          Stream<ContainerInfo> containerInfoStream =
+              scm.getContainerManager().getContainers(state).stream()
+                .filter(info -> info.containerID().getId() >= startContainerID)
+                //Filtering EC replication type as EC will not have factor.
+                .filter(info -> info
+                    .getReplicationType() != HddsProtos.ReplicationType.EC)
+                .filter(info -> (info.getReplicationFactor() == factor))
+                .sorted();
+          List<ContainerInfo> containerInfos = containerInfoStream.collect(
+              Collectors.toList());
+          return Pair.of(containerInfos.stream().limit(count).collect(Collectors.toList()),
```

Comment:
This is repeated on line 487 you could avoid it by returning outside of the if-else block.

Reply:
yeah! refactored two listContainer methods, to make them reuse common code as much as possible.